core/src/main/java/pl/agh/edu/engine/time/command/NRepeatingTimeCommand.java

```diff
@@ -2,19 +2,54 @@
 
 import java.time.LocalDateTime;
 
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.Serializer;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import com.esotericsoftware.kryo.serializers.ClosureSerializer;
+
 import pl.agh.edu.engine.time.Frequency;
+import pl.agh.edu.serialization.KryoConfig;
 
 public class NRepeatingTimeCommand extends RepeatingTimeCommand {
-	private final Runnable toExecuteAfterLastRepetition;
+	private final SerializableRunnable toExecuteAfterLastRepetition;
 	private long counter;
 
-	public NRepeatingTimeCommand(Frequency frequency, Runnable toExecute, LocalDateTime dueTime, long N, Runnable toExecuteAfterLastRepetition) {
+	static {
+		KryoConfig.kryo.register(NRepeatingTimeCommand.class, new Serializer<NRepeatingTimeCommand>() {
+			@Override
+			public void write(Kryo kryo, Output output, NRepeatingTimeCommand object) {
+				kryo.writeObject(output, object.frequency);
+				kryo.writeObject(output, object.toExecute);
+				kryo.writeObject(output, object.dueDateTime);
+				kryo.writeObject(output, object.counter);
+				kryo.writeObject(output, object.toExecuteAfterLastRepetition);
+				kryo.writeObject(output, KryoConfig.getPrivateFieldValue(object, "version", Long.class));
+				kryo.writeObject(output, object.toStop);
+			}
+
+			@Override
+			public NRepeatingTimeCommand read(Kryo kryo, Input input, Class<? extends NRepeatingTimeCommand> type) {
+				NRepeatingTimeCommand nRepeatingTimeCommand = new NRepeatingTimeCommand(
+						kryo.readObject(input, Frequency.class),
+						(SerializableRunnable) kryo.readObject(input, ClosureSerializer.Closure.class),
+						kryo.readObject(input, LocalDateTime.class),
+						kryo.readObject(input, Long.class),
+						(SerializableRunnable) kryo.readObject(input, ClosureSerializer.Closure.class));
+				KryoConfig.setPrivateFieldValue(nRepeatingTimeCommand, "version", kryo.readObject(input, Long.class));
```

Comment:
IMHO w przypadkach takich jak ten z użyciem pól prywatnych czy tam final stworzył bym nowy prywatny konstruktor, ale decyzja up to you

Reply:
czekaj zaraz to wrzuce xd

Reply:
już wrzucone